src/raster/r.fusion/r.fusion.py

```diff
@@ -0,0 +1,371 @@
+# %  options: difference,proportion
+# %  answer: difference
+# %  required: no
+# %end
+
+# %option G_OPT_R_OUTPUT
+# %end
+
+# %flag
+# %  key: r
+# %  label: Restrict output value range to input value range
+# %  description: Recommended for imagery with a defined valid value range, e.g. [0, 255]
+# %end
+
+
+import atexit
+import os
+import sys
+
+import grass.script as gscript
+
+# initialize global vars
+rm_rasters = []
+
+
+def cleanup():
+    nuldev = open(os.devnull, "w")
+    kwargs = {"flags": "f", "quiet": True, "stderr": nuldev}
+
+    # remove temporary raster maps
+    for rmrast in rm_rasters:
+        if gscript.find_file(name=rmrast, element="cell")["file"]:
+            gscript.run_command("g.remove", type="raster", name=rmrast, **kwargs)
+
+
+def main():
+    lowres_map = options["input"]
+    highres_map = options["highres"]
+    output_map = options["output"]
+    fusion_method = options["method"]
+    fit_range = flags["r"]
+
+    gscript.use_temp_region()
+    current_region = gscript.region()
+
+    # get range of lowres raster
+    gscript.run_command("g.region", align=lowres_map)
+    statslr = gscript.parse_command(
+        "r.univar",
+        map=lowres_map,
+        flags="g",
+        parse=(gscript.parse_key_val, {"sep": "="}),
+    )
+    # get info of lowres raster
+    infolr = gscript.raster_info(lowres_map)
+
+    # get range of highres raster
+    gscript.run_command(
+        "g.region",
+        w=current_region["w"],
+        s=current_region["s"],
+        e=current_region["e"],
+        n=current_region["n"],
+        ewres=current_region["ewres"],
+        nsres=current_region["nsres"],
+        quiet=True,
+    )
+    gscript.run_command("g.region", align=highres_map)
+    statshr = gscript.parse_command(
+        "r.univar",
+        map=highres_map,
+        flags="g",
+        parse=(gscript.parse_key_val, {"sep": "="}),
+    )
+    # get info of highres raster
+    infohr = gscript.raster_info(highres_map)
+
+    lrmin = float(statslr["min"])
+    lrmax = float(statslr["max"])
+    hrmin = float(statshr["min"])
+    hrmax = float(statshr["max"])
+
+    # check input for method proportion
+    if fusion_method == "proportion":
+        if hrmin <= 0:
+            gscript.fatal(
+                "High resolution map must have only positive values for method 'proportion'"
+            )
+        if lrmin < 0:
+            gscript.fatal(
+                "Low resolution map must have only non-geative values for method 'proportion'"
+            )
+
+    last_result = highres_map
+
+    if fusion_method == "difference":
+        gscript.message(_("Scaling highres map to range of lowres map..."))
+        # rescale highres raster to lowres raster
+        scale_factor = (lrmax - lrmin) / (hrmax - hrmin)
+        # formula: rescaled = (highres - hrmin) * scale_factor + lrmin
+        highres_map_scaled = f"highres_map_scaled_tmp_{os.getpid()}"
```

Comment:
Just a suggestion, there is [append_node_pid](https://grass.osgeo.org/grass83/manuals/libpython/script.html#script.utils.append_node_pid) function in library for these cases, which makes this safer for running on HPC.

Reply:
Thanks for the hint, I have changed generation of tmp names accordingly.